core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java

```diff
@@ -307,24 +308,48 @@ protected enum CommitStatus {
    * @param newMetadataLocation the path of the new commit file
    * @param config metadata to use for configuration
    * @return Commit Status of Success, Failure or Unknown
+   * @deprecated since 1.5.0, use {@link BaseMetastoreTableOperations#checkCommitStatus(String,
+   *     String, Map, Function)} instead.
    */
+  @Deprecated
```

Comment:
There's no need to deprecate this method and create a public method in its place. It introduces unnecessary changes in other catalogs only to have the same call pattern everywhere.

If you want to use the logic elsewhere, then the new public method shouldn't live here. I recommend creating a static util method somewhere else.

Reply:
@nk1506: Please remove deprecation as per @rdblue comment.